lib/Feature/FeatureModelParser.cpp

```diff
@@ -221,7 +255,11 @@ std::unique_ptr<xmlDoc, void (*)(xmlDocPtr)> FeatureModelXmlParser::parseDoc() {
   if (Doc && Ctxt->valid) {
     xmlValidateDtd(&Ctxt->vctxt, Doc.get(), createDtd().get());
     if (Ctxt->vctxt.valid) {
-      return Doc;
+      if (xmlValidateDtdFinal(&Ctxt->vctxt, Doc.get()) == 1) {
+        return Doc;
+      }
+      llvm_unreachable("boink");
```

Comment:
```suggestion
      llvm_unreachable("Failed to validate DTD in final step");
```
We should use a good error message here :laughing: